fe2-android/app/src/main/java/com/github/dedis/popstellar/di/DataRegistryModule.java

```diff
@@ -0,0 +1,141 @@
+import java.util.Map;
+
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import dagger.hilt.InstallIn;
+import dagger.hilt.components.SingletonComponent;
+
+@Module
+@InstallIn(SingletonComponent.class)
+public abstract class DataRegistryModule {
+
+  private DataRegistryModule() {}
+
+  @Provides
+  @Singleton
+  public static DataRegistry provideDataRegistry() {
+    Map<DataRegistry.EntryPair, Pair<Class<? extends Data>, DataHandler<? extends Data>>> map =
+        new HashMap<>();
+
+    // Lao
+    add(map, LAO, CREATE, CreateLao.class, LaoHandler::handleCreateLao);
+    add(map, LAO, UPDATE, UpdateLao.class, LaoHandler::handleUpdateLao);
+    add(map, LAO, STATE, StateLao.class, LaoHandler::handleStateLao);
+
+    // Meeting
+    add(map, MEETING, CREATE, CreateMeeting.class, null);
+    add(map, MEETING, STATE, StateMeeting.class, null);
+
+    // Message
+    add(map, MESSAGE, WITNESS, WitnessMessageSignature.class, null);
+
+    // Roll Call
+    add(map, ROLL_CALL, CREATE, CreateRollCall.class, RollCallHandler::handleCreateRollCall);
+    add(map, ROLL_CALL, OPEN, OpenRollCall.class, RollCallHandler::handleOpenRollCall);
+    add(map, ROLL_CALL, REOPEN, OpenRollCall.class, RollCallHandler::handleOpenRollCall);
+    add(map, ROLL_CALL, CLOSE, CloseRollCall.class, RollCallHandler::handleCloseRollCall);
+
+    // Election
+    add(map, ELECTION, SETUP, ElectionSetup.class, ElectionHandler::handleElectionSetup);
+    add(map, ELECTION, CAST_VOTE, CastVote.class, ElectionHandler::handleCastVote);
+    add(map, ELECTION, END, ElectionEnd.class, ElectionHandler::handleElectionEnd);
+    add(map, ELECTION, RESULT, ElectionResult.class, ElectionHandler::handleElectionResult);
+
+    // Consensus
+    add(map, CONSENSUS, ELECT, ConsensusElect.class, ConsensusHandler::handleElect);
+    add(
+        map,
+        CONSENSUS,
+        ELECT_ACCEPT,
+        ConsensusElectAccept.class,
+        ConsensusHandler::handleElectAccept);
+    add(map, CONSENSUS, PREPARE, ConsensusPrepare.class, ConsensusHandler::handleBackend);
+    add(map, CONSENSUS, PROMISE, ConsensusPromise.class, ConsensusHandler::handleBackend);
+    add(map, CONSENSUS, PROPOSE, ConsensusPropose.class, ConsensusHandler::handleBackend);
+    add(map, CONSENSUS, ACCEPT, ConsensusAccept.class, ConsensusHandler::handleBackend);
+    add(map, CONSENSUS, LEARN, ConsensusLearn.class, ConsensusHandler::handleLearn);
+
+    // Social Media
+    add(map, CHIRP, ADD, AddChirp.class, ChirpHandler::handleChirpAdd);
+    add(map, CHIRP, ADD_BROADCAST, AddChirpBroadcast.class, null);
+
+    return new DataRegistry(map);
+  }
+
+  private static <T extends Data> void add(
+      Map<DataRegistry.EntryPair, Pair<Class<? extends Data>, DataHandler<? extends Data>>> map,
+      Objects obj,
+      Action action,
+      Class<T> dataClass,
+      DataHandler<T> dataHandler) {
+    map.put(DataRegistry.pair(obj, action), Pair.create(dataClass, dataHandler));
+  }
```

Comment:
Very good idea to register the datas in the Registry module !

But that that point, I think we should create a Builder for the registry rather than passing it a Map.
This Map signature is starting to be quite long, and it won't get better in the future.

I will do in during the day if I am tired of my wallet refactor.